docker-compose.yml

```diff
@@ -1,4 +1,4 @@
-version: '3'
+#version: '3'
```

Comment:
Not specifying a version seems to make Docker Compose deal with it as if it's the new compose-spec, which kind of marries Docker Compose v2 and v3, and is supported since Compose v1.27.0+.

https://github.com/compose-spec/compose-spec/blob/master/spec.md#version-top-level-element

> The latest and recommended version of the Compose file format is defined by the Compose Specification. This format merges the 2.x and 3.x versions and is implemented by Compose 1.27.0+.
>
> https://docs.docker.com/compose/compose-file/compose-versioning/#versioning

If your local docker-compose is of that version or greater, then I bet that your docker-compose is using the new compose-spec and not docker-compose v2. And compose-spec does support the depends_on with conditions, while docker-compose v3 seems to not support it.

What's more interesting is that they've renamed the file to separate from Docker by now calling it just `compose.yml` (https://github.com/compose-spec/compose-spec/blob/master/spec.md#compose-file).

However, the compose-spec does not seem to support the `docker-compose.override.yml` that we take advantage of. Maybe Docker Compose still does support it, but it's not written in the compose spec anyway. What is written there though is the new [`extends`](https://github.com/compose-spec/compose-spec/blob/master/spec.md#extends) field which can cherry-pick a service from a different file. However&times;2: The `extends` seems to not be allowed to be used on base services which depends on other services, where `depends_on` is one of these cases. Could perhaps be worked around still.

---

In my opinion, jumping onboard of this compose-spec seems good, supporting the new upcoming standard which is more container implementation agnostic.

So the changes I would personally want to see to make this happen would be:

- Remove the commented out `#version: '3'` altogether.
- Remove the `version: '3'` in `docker-compose.build.yml` as well.
- Rename the file to simply `compose.yml` instead.
- Perhaps create a base compose file with just all of the services defined
- Change this file to reference the base compose file on all services
- Rename the `docker-compose.build.yml` to `compose.build.yml`
- Change the `compose.build.yml` to also reference the base compose file
- In both `compose.yml` and `compose.build.yml`, add in the `depends_on` back again, as the base service cannot have a `depends_on`.

What do you think?

Reply:
I believe Docker Compose v3.8+ supports `depends_on` the way we need to use it. I was using it earlier for something else.